src/ucp/rndv/rndv.c

```diff
@@ -20,6 +20,32 @@
 #include <ucp/proto/proto_am.inl>
 #include <ucs/datastruct/queue.h>
 
+static UCS_F_ALWAYS_INLINE int
+ucp_rndv_memtype_direct_support(ucp_context_h context, size_t reg_length,
+                                ucs_memory_type_t mem_type,
+                                uint64_t reg_mem_types)
+{
+    return ((mem_type == UCS_MEMORY_TYPE_HOST) ||
+            !(reg_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST)) ||
+            (reg_length < context->config.ext.rndv_memtype_direct_size));
```

Comment:
why need so many checks (which add cpu overhead), can't just check `reg_length < context->config.ext.rndv_memtype_direct_size` and `(mem_type == UCS_MEMORY_TYPE_HOST)` in ucp_rndv_is_get_zcopy?

Reply:
we wantt another check do only for GDR lane and not on cuda-ipc lane

Reply:
how is GDR lane differentiated from cuda-ipc lane?

Reply:
cuda-ipc lane can't register HOST memory. assuming that only GDR lane can do HOST and MEMTYPE registration

Reply:
1. it's very non-obvious (and not documented). can we check by sys_dev instead? for example if they have same sysdev than it's ok to use zcopy
2. in general need to reduce the overhead of this check: use bitmap operations, precompute into ep_config whatever possible, etc

Reply:
@yosefe As per our discussion, I added some verbose and mem type check in the function